doc/api/buffer.md

```diff
@@ -3515,6 +3515,482 @@ console.log(buf.toString());
+* Returns: {string}
+
+Decodes buf to a string according to UCS-2 character encoding. `start` and
+`end` may be passed to decode only a subset of `buf`.
+
+```mjs
+import { Buffer } from 'node:buffer';
+const buf = Buffer.from('buffer');
+
+// Equivalent to buf.toString('ucs2', 0, 3)
+console.log(buf.ucs2Slice(0, 3));
+// Prints: 畢
+```
+
+```cjs
+const { Buffer } = require('node:buffer');
+const buf = Buffer.from('buffer');
+
+// Equivalent to buf.toString('ucs2', 0, 3)
+console.log(buf.ucs2Slice(0, 3));
+// Prints: 畢
+```
+
+### `buf.utf8Slice([start[, end]])`
+
+* `start` {integer} The byte offset to start slicing at. **Default**: 0.
+* `end` {integer} The byte offset to stop slicing at (not inclusive).
+  **Default**: `buf.length`.
+* Returns: {string}
+
+Decodes buf to a string according to UTF-8 character encoding. `start` and
+`end` may be passed to decode only a subset of `buf`.
+
+```mjs
+import { Buffer } from 'node:buffer';
+const buf = Buffer.from('buffer');
+
+// Equivalent to buf.toString('utf8', 0, 3)
+console.log(buf.utf8Slice(0, 3));
+// Prints: buf
+```
+
+```cjs
+const { Buffer } = require('node:buffer');
+const buf = Buffer.from('buffer');
+
+// Equivalent to buf.toString('utf8', 0, 3)
+console.log(buf.utf8Slice(0, 3));
+// Prints: buf
+```
+
+### `buf.asciiWrite(string[, offset[, length]])`
+
+* `string` {string} String to write to `buf`.
+* `offset` {integer} Number of bytes to skip before starting to write
+  `string`. **Default**: 0.
+* `length` {integer} Maximum number of bytes to write. **Default**: `buf.
+  length - offset`.
+
+Writes `string` to `buf` at `offset` according to the ASCII character encoding
+and returns the number of bytes written. If `buf` did not contain enough space
+to fit the entire string, only part of `string` will be written. However,
+partially encoded characters will not be written.
+
+In most cases, `buf.write` is preferable, especially if `encoding` is variable.
+
+```mjs
+import { Buffer } from 'node:buffer';
+const buf = Buffer.alloc(10);
+
+const bytesWritten = buf.asciiWrite('buffer');
+console.log(bytesWritten);
+// Prints: 6
+```
+
+```cjs
+const { Buffer } = require('node:buffer');
+const buf = Buffer.alloc(10);
+
+const bytesWritten = buf.asciiWrite('buffer');
+console.log(bytesWritten);
+// Prints: 6
+```
+
+### `buf.base64Write(string[, offset[, length]])`
+
+<!-- YAML
+added: v0.11.3
+-->
+
+* `string` {string} String to write to `buf`.
+* `offset` {integer} Number of bytes to skip before starting to write
+  `string`. **Default**: 0.
+* `length` {integer} Maximum number of bytes to write. **Default**: `buf.
+  length - offset`.
+* Returns: {integer} The number of bytes written.
+
+Writes `string` to `buf` at `offset` according to the base64 character encoding
+and returns the number of bytes written. If `buf` did not contain enough space
+to fit the entire string, only part of `string` will be written. However,
+partially encoded characters will not be written.
```

Comment:
I'm not sure I'd know what to expect if the input string contains non-ASCII chars, does it first convert to UTF-8 or some other encoding then convert it to base64?